core/new-gui/src/app/workspace/service/drag-drop/drag-drop.service.ts

```diff
@@ -0,0 +1,214 @@
+    jQuery('#' + dropElementID).droppable({
+      drop: (event: any, ui) => this.handleOperatorDrop(event, ui)
+    });
+  }
+
+  /**
+   * Creates a DOM Element that visually looks identical to the operator when dropped on main workflow editor
+   *
+   * This function temporarily creates a DOM element which contains a JointJS paper that has the exact size of the operator,
+   *    then create the operator Element based on the operatorType and make it fully occupy the JointJS paper.
+   *
+   * The temporary JointJS paper element has ID "flyingJointPaper". This DOM elememtn will be destroyed by jQueryUI when the dragging ends.
+   *
+   * @param operatorType - the type of the operator
+   */
+  private createFlyingOperatorElement(operatorType: string): JQuery<HTMLElement> {
+    this.currentOperatorType = operatorType;
+
+    // create a temporary ghost element
+    jQuery('body').append('<div id="flyingJointPaper" style="position:fixed;z-index:100;pointer-event:none;"></div>');
+
+    // create an operator and get the UI element from the operator type
+    const operator = this.workflowUtilService.getNewOperatorPredicate(operatorType);
+    const operatorUIElement = this.jointUIService.getJointOperatorElement(operator, { x: 0, y: 0 });
+
+    // create the jointjs model and paper of the ghost element
+    const tempGhostModel = new joint.dia.Graph();
+    const tempGhostPaper = new joint.dia.Paper({
+      el: jQuery('#flyingJointPaper'),
+      width: JointUIService.DEFAULT_OPERATOR_WIDTH,
+      height: JointUIService.DEFAULT_OPERATOR_HEIGHT,
+      model: tempGhostModel,
+    });
+
+    // add the operator JointJS element to the paper
+    tempGhostModel.addCell(operatorUIElement);
+
+    // return the jQuery object of the DOM Element
+    return jQuery('#flyingJointPaper');
+  }
+
+  /**
+   * Hanlder function for jQueryUI's drag started event.
+   * It converts the event to the drag started Subject.
+   *
+   * @param event JQuery.Event type, although JQueryUI typing says the type is Event, the object's actual type is JQuery.Event
+   * @param ui jQueryUI Draggable Event UI
+   */
+  private handleOperatorStartDrag(event: JQuery.Event, ui: JQueryUI.DraggableEventUIParams): void {
+    const eventElement = event.toElement;
+    if (eventElement === undefined) {
+      throw new Error('drag and drop: cannot find element when drag is started');
+    }
+    // get the operatorType based on the DOM element ID
+    const operatorType = this.elementOperatorTypeMap.get(eventElement.id);
+    if (operatorType === undefined) {
+      throw new Error(`drag and drop: cannot find operator type ${operatorType} from DOM element ${eventElement}`);
+    }
+    // set the currentOperatorType
+    this.currentOperatorType = operatorType;
+    // notify the subject of the event
+    this.operatorDragStartedSubject.next({operatorType});
+  }
+
+  /**
+   * Hanlder function for jQueryUI's drag started event.
```

Comment:
drag stopped event. 
Also, is last line incomplete?

Reply:
fixed